libsolutil/JSON.cpp

```diff
@@ -32,9 +32,12 @@
 
 using namespace std;
 
+// This check is not very useful now, since 1.9.4 had a bug where the STRING was updated, but not the PATCH.
+// The release version: https://github.com/open-source-parsers/jsoncpp/blob/1.9.4/include/json/version.h
+// The fix post-release: https://github.com/open-source-parsers/jsoncpp/issues/1224
 static_assert(
 	(JSONCPP_VERSION_MAJOR == 1) && (JSONCPP_VERSION_MINOR == 9) && (JSONCPP_VERSION_PATCH == 3),
```

Comment:
Despite the comment, I am not completely sure why you didn't update the `3`?
We are checking for `JSONCPP_VERSION_PATCH == 3` because upstream didn't update that variable?
Wouldn't it be better to just remove the check then? Or  at least comment?
Or you could for check for both, 3 and 4 I guess, in case they do update it eventually?

Reply:
Isn't the first line saying that? :)

Reply:
> Wouldn't it be better to just remove the check then? Or at least comment?
> Or you could check for both, 3 and 4 I guess, in case they do update it eventually? But that would probably then become patch nr 5.

Checking for both 3 and 4 sounds good but this is only a one-time thing for 1.9.4 so maybe it's not worth bothering. This will be back to normal in 1.9.5 and we'll be able to remove the comment anyway.

Reply:
BTW, I just realized that it's only because of this bug that #11045 even got reported. It was a problem with compiling using 1.9.4 and we did not even support that version at that time but the check did not catch that which gave me an impression that we did.